shakenfist/baseobject.py

```diff
@@ -93,11 +95,27 @@ def update_state(self, new_value, error_message=None):
                     'Invalid state change from %s to %s for object=%s uuid=%s',
                     orig.value, new_value, self.object_type, self.uuid)
 
-            new_state = State(new_value, time.time(), error_message)
+            new_state = State(new_value, time.time())
             self._db_set_attribute('state', new_state)
             self.add_event('state changed',
                            '%s -> %s' % (orig.value, new_value))
 
+    @property
+    def error_msg(self):
+        db_data = self._db_get_attribute('error_msg')
+        if not db_data:
+            return None
+        return db_data
+
+    @error_msg.setter
+    def error_msg(self, msg):
+        s = self.state
+        if s.value != 'error':
+            raise exceptions.InvalidStateException(
+                'Object not in error state (state=%s, object=%s)' % (
+                    s, self.object_type))
+        self._db_set_attribute('error_msg', msg)
```

Comment:
Additionally (this was more obvious once I read the upgrade), this would be the only example of storing just a string at a key. Should we use a dict here just in case we want to add other fields later?

Reply:
Sure.